js/entity_browser.entity_reference.js

```diff
@@ -28,6 +40,24 @@
     entity_ids = entity_ids + $.map(entities, function(item) {return item[0]}).join(' ');
     $(this).parent().parent().find('input[type*=hidden]').val(entity_ids);
     $(this).parent().parent().find('input[type*=hidden]').trigger('entity_browser_value_updated');
-  }
+  };
+
+  /**
+   * Reacts on sorting of the entities.
+   *
+   * @param event
+   *   Event object.
+   * @param ui
+   *   Object with detailed information about the sort event.
+   */
+  Drupal.entityBrowserEntityReference.entitiesReordered = function(event, ui) {
+    var items = $(this).find('li');
+    var ids = [];
+    for (var i = 0; i < items.length; i++) {
+      ids[i] = $(items[i]).attr('data-entity-id');
+    }
+
+    $(this).parent().parent().find('input[type*=hidden]').val(ids.join(' '));
```

Comment:
I don't think so, but I am not JS expert. There is only one hidden input element in this field widget so I guess should be ok.